src/controller/RacingGameController.js

```diff
@@ -0,0 +1,72 @@
+const RacingGame = require('../model/RacingGame');
+const inputView = require('../view/inputView');
+const outputView = require('../view/outputView');
+const validate = require('../utils/validation');
+const { SIGN, ERROR } = require('../utils/constants');
+
+class RacingGameController {
+  #inputView;
+  #outputView;
+  constructor() {
+    this.#inputView = inputView;
+    this.#outputView = outputView;
+    this.initiate();
+  }
+
+  initiate() {
+    this.#readCarNames();
+  }
+
+  #readCarNames() {
+    this.#inputView.readUserInputNames((input) => {
+      const names = input.split(SIGN.comma);
+      this.#handleCarNamesException(names);
+    });
+  }
+
+  #readTrialCounts(game) {
+    this.#inputView.readUserInputTrialCounts((input) => {
+      this.#handleTrialCountsException(input, game);
+    });
+  }
+
+  #handleCarNamesException(names) {
+    try {
+      this.#validateCarNames(names);
+      const racingGame = new RacingGame(names);
+      this.#readTrialCounts(racingGame);
+    } catch (error) {
+      this.#outputView.printError(ERROR.userInputNameInvalid);
+      this.#readCarNames();
+    }
+  }
+
+  #validateCarNames(names) {
+    for (const item of names) {
+      validate.userInputName(item);
+    }
+  }
+
+  #handleTrialCountsException(input, game) {
+    try {
+      const number = Number(input);
+      validate.userInputNumber(number);
+      this.#handleTrialCounts(number, game);
+    } catch (error) {
+      this.#outputView.printError(ERROR.userInputNumberInvalid);
+      this.#readTrialCounts(game);
+    }
+  }
```

Comment:
이러면 handleTrialCountsException가 제대로 동작하는지 테스트할 수 없을 것 같네.
이름만 봐서는 trialCountInput만 인자로 받을 거 같은데 game도 함께 받고 있어서 테스트할 때 불필요하게 game도 만들어줘야 하니까.

async await을 못 써서 어쩔 수 없이 이렇게 한거겠지? 그래도 이게 좋은 방식이 아니란 건 알아둬야 돼. 왜 좋지 않은지도 알아둬야 하고. 나는 지금 방법이 안떠오르지만 한번 찾아보는 것도 나쁘지 않아. 내가 모르는 방식도 있을 수 있으니까. 
근데 굳이 당장 찾을 필요는 없어. 어차피 최종 코테에서는 그냥 완성이 중요하고 돌아가는게 중요하니까.

Reply:
```suggestion
  #handleTrialCountsException(input, game) {
    try {
      const trialCount = Number(input);
      this.#validateTrialCountInput(trialCount); // 1
      
      this.#handleTrialCounts(number, game);
    } catch (error) {
      this.#outputView.printError(ERROR.userInputNumberInvalid);
      this.#readTrialCounts(game);
    }
  }
  
 #validateTrialCountInput(number){
   validate.userInputNumber(number);
 }
```
1. 여기서는 또 validate를 require 한거 바로 쓰고 있네. validate.userInputNumber가 단순히 숫자인지만 체크하는 거라고 해도 통일성 있게 가는게 가독성에도 좋아.